src/libAtomVM/nifs.c

```diff
@@ -2069,18 +2142,50 @@ static term nif_erlang_atom_to_list_1(Context *ctx, int argc, term argv[])
         RAISE_ERROR(OUT_OF_MEMORY_ATOM);
     }
 
-    if (UNLIKELY(memory_ensure_free_opt(ctx, atom_len * 2, MEMORY_CAN_SHRINK) != MEMORY_GC_OK)) {
+    atom_table_write_bytes(ctx->global->atom_table, atom_ref, atom_len, atom_buf);
+
+    size_t u8len = unicode_buf_utf8_len((uint8_t *) atom_buf, atom_len);
+    bool latin1 = atom_len == u8len;
+
+    size_t list_len = latin1 ? atom_len : u8len;
+
+    if (UNLIKELY(memory_ensure_free_opt(ctx, list_len * 2, MEMORY_CAN_SHRINK) != MEMORY_GC_OK)) {
```

Comment:
We have a macro for CONS_SIZE in term.h, instead of using 2. We also have a LIST_SIZE macro, if that is preferable.